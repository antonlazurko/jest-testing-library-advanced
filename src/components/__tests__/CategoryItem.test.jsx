import { screen } from "@testing-library/react";
import { renderWithRouter } from "../../utils/testing";
import { CategoryItem } from "../CategoryItem";

describe("CategoryItem", () => {
    test("should render correctly", () => {
        renderWithRouter(<CategoryItem strCategory="test"
            strCategoryThumb="/test.png"
            strCategoryDescription="test"/>);
        expect(screen.getByRole("article")).toMatchSnapshot();
    });
})