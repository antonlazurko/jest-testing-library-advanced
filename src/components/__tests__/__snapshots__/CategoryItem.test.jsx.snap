// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryItem should render correctly 1`] = `
<div
  class="card"
  role="article"
>
  <div
    class="card-image"
  >
    <img
      alt="test"
      src="/test.png"
    />
  </div>
  <div
    class="card-content"
  >
    <span
      class="card-title"
    >
      test
    </span>
    <p>
      test
      ...
    </p>
  </div>
  <div
    class="card-action"
  >
    <a
      class="btn"
      href="/category/test"
    >
      Watch category
    </a>
  </div>
</div>
`;
